### ISSUE TITLE
Error processing environment subsition variable (kc.db) in keycloak.conf

### ISSUE THREAD

**Issue by rpetrescu** [keycloak/keycloak#26041]
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

core

### Describe the bug

I am getting this error while trying to run **kc.sh build**:

0.986 Changes detected in configuration. Updating the server image.
0.994 Invalid value for option **'kc.db': ${KC_DB:mysql}**. Expected values are: dev-file, dev-mem, mariadb, mssql, mysql, oracle, postgres. From ConfigSource PropertiesConfigSource[source=file:/opt/keycloak/bin/../conf/keycloak.conf]

My keycloak.conf file contains many variables which use environment variable substitution, but, apparently, the bug only happens for kc.db variable. If I comment this one, the build command succeeds and the process starts correctly.

### Version

23.0.2, 23.0

### Expected behavior

The build command should succeed. It works fine on version 21.*, 22* (tested with 22.0.3, 22.0.4).

### Actual behavior

The build command should succeed, like it happens on version 21.*, 22.*.

### How to Reproduce?

Add the following variable in keycloak.conf:

db=${KC_DB:mysql}

and run kc.sh build

### Anything else?

I am testing/using this with docker
(FROM quay.io/keycloak/keycloak:23.0.2 as builder)

**Comment by rpetrescu** [comment]
Hello,

I have tried using only KC_DB to override a different setting from
keycloak.conf, but it does not work. More explicitly:
- in keycloak.conf I have "db=postgres", then in .env I have added
"KC_DB=mysql", I have rebuilt the images with docker-compose, but the env
variable is not picked up and KC does not start.

It is definitely a regression, because it works fine in 22.* and it has
happened before and closed (only those time for other variables):
https://github.com/keycloak/keycloak/issues/20609
https://github.com/keycloak/keycloak/issues/20032

Thank you

**Comment by shawkins** [comment]
@rpetrescu @sschu this works again in main because of the changes for #24668 - a backport is possible, but it needs a selective amount of those changes. 

cc @vmuzikar 

>  but the env variable is not picked up and KC does not start.

I can't reproduce this behavior. The env takes precedence over the keycloak.conf

**Comment by shawkins** [comment]
@sschu @vmuzikar any vote on whether to do a backport, or can this just be closed?

**Comment by sschu** [comment]
@shawkins I would say there is a valid workaround for 23 by just using environment variables directly so if the backport is not trivial, I wouldn't do it.

**Comment by vmuzikar** [comment]
I agree with @sschu. This will be implicitly fixed in 24 which is just around the corner.

**Comment by rpetrescu-epo** [comment]
Indeed, the workaround is available, but there cases with a lot of environments (20+) and we would need to update the configurations everywhere.
But I agree, if v24 will be released soon and the backport is not trivial, then we can skip 23 and go directly with 24. 

Thank you.
